compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -852,7 +852,7 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
           val evidence = evTyper.inferImplicitArg(tpt.tpe, tpt.span)(using evCtx)
```

Comment:
You could wrap the whole block using `inContext(evCtx) { ... }` to make sure we don't accidentally use the wrong context anywhere.

Reply:
Fixed